statistics/updateDatabase.js

```diff
@@ -0,0 +1,222 @@
+        }
+    }
+    return courses;
+}
+
+function exportDataFromSingleSpreadsheet(path, format, courses) {
+
+    console.log(`Reading spreadsheet ${__dirname}/${path}...`);
+    const worksheet = xlsx.parse(`${__dirname}/${path}`);
+
+    //Depending on the format of the document, slice away the first worksheet.
+    worksheet.slice(format).forEach((ws) => {
+
+        const data = ws.data.slice(1);
+        //For each row in the data from the 2nd onwards
+        data.forEach((r) => {
+            //Course Code, Course Name, ex: TKTFY, ex: Teknisk Fysik, Examination ID number, Type of examination, Course examination moment HP, date, grade, amount, fraction of total 
+            let courseCode, courseName, programShort, programLong, resultId, type, points, date, grade, count, fraction;
+            [courseCode, courseName, programShort, programLong, resultId, type, points, date, grade, count, fraction] = r;
+            if (date.toString().length === 5) { // Excel short date format
+                date = (new Date((date - (25567 + 2)) * 24 * 60 * 60 * 1000)).toISOString().substr(0, 10);
+            } else {
+                date = new Date(date + ' 12:00').toISOString().substr(0, 10);
+            }
+
+            if (!(courseCode in courses)) { //If a row with a new course appears
+                let course = new Course();
+                courseProperties.forEach((p, i) => {
+                    course[p] = r[i];
+                });
+
+                let result = new Result();
+                result.resultId = resultId;
+                result.date = date;
+                result.type = type;
+                result[grade] = count;
+                course.results = [result];
+                courses[courseCode] = course;
+            } else { //A row with a previously seen course appears
+                let course = courses[courseCode];
+                let results = course.results;
+                let result = results[results.length - 1];
+                if (date === result.date) {
+                    result[grade] = count;
+                    course.results[course.results.length - 1] = result;
+                } else {
+                    let result = new Result();
+                    result.resultId = resultId;
+                    result.date = date;
+                    result.type = type;
+                    result[grade] = count;
+                    course.results.push(result);
+                }
+                courses[courseCode] = course;
+            }
+        });
+    });
+
+    return courses
+}
+
+//Main Async function
+async function main() {
+    mongoose.set('useCreateIndex', true);
+    mongoose.set('useNewUrlParser', true);
+    console.log("Connecting to the database...");
+    await mongoose.connect('mongodb://localhost/stats', {
+        useNewUrlParser: true,
+        useUnifiedTopology: true
+    });
```

Comment:
It would be nice if the database connection creation could use the dotenv configuration instead of beeing hard coded.

Reply:
Might also be nice to have an option to drop the database before updating. This functionality could then be used e.g. by the webhook so it does not need to keep track of which database to drop before updating.

Reply:
I copied it from the old addFields.js which use hardcoded path. Don't know ihow/f dotenv works but can approve a pull request if someone else makes it work😉

Reply:
I will work on a pull request